src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/ModifyQueueCapacityAction.java

```diff
@@ -23,34 +23,43 @@
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.grpc.ResourceEnum;
 import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
 
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.util.NodeConfigCacheReaderUtil;
 import java.util.Collections;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class ModifyQueueCapacityAction extends SuppressibleAction {
 
-  public static final String NAME = "modify_queue_capacity";
-  public static final long COOL_OFF_PERIOD_IN_MILLIS = 300 * 1_000;
-
-  private int currentCapacity;
-  private int desiredCapacity;
-  private ResourceEnum threadPool;
-  private NodeKey esNode;
-
-  private Map<ResourceEnum, Integer> lowerBound = new HashMap<>();
-  private Map<ResourceEnum, Integer> upperBound = new HashMap<>();
-
-  public ModifyQueueCapacityAction(NodeKey esNode, ResourceEnum threadPool, int currentCapacity,
-      boolean increase, AppContext appContext) {
+  private static final Logger LOG = LogManager.getLogger(ModifyQueueCapacityAction.class);
+  public static final String NAME = "ModifyQueueCapacity";
+
+  private final ResourceEnum threadPool;
+  private final NodeKey esNode;
+  private final int desiredCapacity;
+  private final int currentCapacity;
+  private final int lowerBound;
+  private final int upperBound;
```

Comment:
Why do we want lower and upper bound here? Builder takes these values in consideration for calculating the desired value

Reply:
one use case is the decider might use this upper/lower bound to calculate the proper step size for each action. We don't use it at this moment but let's say JVM decider will bucketize the current capacity between the [lower, upper] range and based the bucket this resource is allocated, the decider might choose different step size to tune the resource

Reply:
we can remove them later if decider is end up not using it